controllers/authrealm/dex.go

```diff
@@ -396,12 +397,26 @@ func (r *AuthRealmReconciler) createDexConnectors(authRealm *identitatemv1alpha1
 			r.Log.Info("generated intermediate connectors", "cs", cs)
 		case openshiftconfigv1.IdentityProviderTypeLDAP:
 			r.Log.Info("create connector for LDAP")
+			url, err := ldaputil.ParseURL(idp.LDAP.URL)
+			if err != nil {
+				return nil, fmt.Errorf("Error parsing LDAP URL: %v", err)
+			}
+			var scope string
+			switch url.Scope {
+			case ldaputil.ScopeSingleLevel:
+				scope = "one"
+			case ldaputil.ScopeWholeSubtree:
+				scope = "sub"
+			case ldaputil.ScopeBaseObject:
+				scope = "base"
```

Comment:
The OpenShift docs say they only support "one" or "sub", as does Dex. See https://github.com/dexidp/dex/blob/9d3471e39b24c11350b8c911fc6e736f0be7c785/connector/ldap/ldap.go#L174

What does OpenShift do if you put a scope of base? Seems we should not allow it.

Reply:
I'll remove base.